plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidDrawAllocationRefactoring.java

```diff
@@ -0,0 +1,170 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program under LICENSE-GNUGPL.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution under LICENSE-ECLIPSE, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.autorefactor.refactoring.rules;
+
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ASTVisitor;
+import org.eclipse.jdt.core.dom.CastExpression;
+import org.eclipse.jdt.core.dom.ClassInstanceCreation;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.SimpleType;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.core.dom.VariableDeclarationStatement;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+
+import static org.autorefactor.refactoring.ASTHelper.*;
+import static org.eclipse.jdt.core.dom.ASTNode.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.autorefactor.refactoring.ASTBuilder;
+import org.autorefactor.refactoring.Refactorings;
+
+/** See {@link #getDescription()} method. */
+public class AndroidDrawAllocationRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        return "Optimization for Android applications to avoid the allocation of"
+                + " objects inside drawing routines. ";
+    }
+
+    @Override
+    public String getName() {
+        return "Android DrawAllocation";
+    }
+
+    @Override
+    public boolean visit(MethodDeclaration node) {
+        if (isMethod(node, "android.widget.TextView", "onDraw", "android.graphics.Canvas")) {
+            final ASTBuilder b = this.ctx.getASTBuilder();
+            final Refactorings r = this.ctx.getRefactorings();
+            node.accept(new OnDrawTransformer(this.ctx, node));
+        }
+        return VISIT_SUBTREE;
+    }
+
+    static class OnDrawTransformer extends ASTVisitor {
+        private RefactoringContext ctx;
+        private MethodDeclaration onDrawDeclaration;
+
+        public OnDrawTransformer(RefactoringContext ctx, MethodDeclaration onDrawDeclaration) {
+            this.ctx = ctx;
+            this.onDrawDeclaration = onDrawDeclaration;
+        }
+
+        public boolean isMethodBindingSubclassOf(ITypeBinding typeBinding, List<String> superClassStrings) {
+            ITypeBinding superClass = typeBinding;
+            while (superClass != null && !superClass.equals(ctx.getAST().resolveWellKnownType("java.lang.Object"))) {
```

Comment:
Please extract a variable for this out of the loop: `ctx.getAST().resolveWellKnownType("java.lang.Object")`
I suspect doing this repeatedly in the loop is not great performance wise.